src/syntax/ast.rs

```diff
@@ -28,7 +31,6 @@ pub struct Pattern {
 
 #[derive(Debug, PartialEq)]
 pub enum Expression {
-    Pattern(Pattern),
```

Comment:
Yes, sorry, that's exactly what I meant. Sorry, I didn't mean it should be added into the `Expression` enum. Bad placement of my comment.